sky/provision/kubernetes/utils.py

```
@@ -1344,16 +1348,25 @@ def check_credentials(context: Optional[str],
     # `sky launch --gpus <gpu>` and the optimizer does not list Kubernetes as a
     # provider if their cluster GPUs are not setup correctly.
     gpu_msg = ''
-    try:
-        get_accelerator_label_key_value(context,
-                                        acc_type='',
-                                        acc_count=0,
-                                        check_mode=True)
-    except exceptions.ResourcesUnavailableError as e:
-        # If GPUs are not available, we return cluster as enabled (since it can
-        # be a CPU-only cluster) but we also return the exception message which
-        # serves as a hint for how to enable GPU access.
-        gpu_msg = str(e)
+    unlabeled_nodes = get_unlabeled_accelerator_nodes(context)
+    if unlabeled_nodes:
+        gpu_msg = (f'Cluster has {len(unlabeled_nodes)} nodes with '
+                   f'accelerators that are not labeled. '
+                   f'To label the nodes, run '
+                   f'`python -m sky.utils.kubernetes.gpu_labeler '
+                   f'--context {context}`')
+    else:
+        try:
+            get_accelerator_label_key_value(context,
+                                            acc_type='',
+                                            acc_count=0,
+                                            check_mode=True)
```

in what cases will this raise an exception?

---

There are three cases where this function raises an exception:
1. None of the nodes has `nvidia.com/gpu` resource specified in node capacity. It _is_ possible that a user is interacting with a CPU only cluster, so the error printed (which eventually gets translated to a hint) is more of a "double check this is what you intend to do" than "hey you got this wrong"
2. If at least one node has `nvidia.com/gpu` resource but _no_ node in the cluster has a matchable accelerator label. This is a special case of what `get_unlabeled_accelerator_nodes` checks for, which checks if at least one node has `nvidia.com/gpu` resource and at least one node with `nvidia.com/gpu` resource doesn't have a matchable label.
3. At least one node with a matchable accelerator label has an invalid label value.

The PR masks exception #2, since `get_unlabeled_accelerator_nodes` will return a nonempty list of unlabeled node in every case where exception #2 is raised. That said, I didn't directly modify the message in exception #2 because `get_unlabeled_accelerator_nodes` checks for a broader condition.

---

This is extremely helpful. Could you add a comment explaining these different cases? It's otherwise pretty unclear what's going on.